lib/puppet/provider/scheduled_task/win32_taskscheduler.rb

```diff
@@ -426,7 +426,7 @@ def bitfield_from_days(days)
       day = 32 if day == 'last'
 
       integer_day = Integer(day)
-      self.fail "Day must be specified as an integer in the range 1-31, or as 'last'" unless integer_day = day.to_f and integer_day.between?(1,32)
+      self.fail "Day must be specified as an integer in the range 1-31, or as 'last'" unless integer_day.between?(1,32)
```

Comment:
@Iristyle @glennsarti since you're in the task scheduler code, does this make sense? Or were we trying to compare `integer_day == day.to_f`?

Reply:
That is a really hard assertion to read :-( 

I think you're right though.  Looks like it was checking for that, which is useless anyway.